diff --git a/src/lfs.c b/src/lfs.c
index 95ab63b..dca148f 100644
--- a/src/lfs.c
+++ b/src/lfs.c
@@ -285,10 +285,10 @@ static int get_dir(lua_State * L)
   char *path = NULL;
   /* Passing (NULL, 0) is not guaranteed to work.
      Use a temp buffer and size instead. */
-  size_t size = LFS_MAXPATHLEN; /* initial buffer size */
+  int size = LFS_MAXPATHLEN; /* initial buffer size */
   int result;
   while (1) {
-    char *path2 = realloc(path, size);
+    char *path2 = (char *)realloc(path, size);
     if (!path2) {               /* failed to allocate */
       result = pusherror(L, "get_dir realloc() failed");
       break;
@@ -1078,7 +1078,7 @@ static int push_link_target(lua_State * L)
   int tsize, size = 256;        /* size = initial buffer capacity */
   int ok = 0;
   while (!ok) {
-    char *target2 = realloc(target, size);
+    char *target2 = (char *)realloc(target, size);
     if (!target2) {             /* failed to allocate */
       break;
     }
